usaspending_api/common/csv_stream_s3_to_pg.py

```diff
@@ -94,6 +96,50 @@ def _stream_and_copy(
         raise exc
 
 
+def _download_and_copy(
```

Comment:
There is a lot of code duplication between this function and _stream_and_copy. Specifically, all parameters, lines 110-111, lines 119 to 122, lines 125 to 128, and lines 130 to 142 

I get the impression you copy an pasted the stream method and only changed the resources that are being managed.

For example, it appears the key differences between these methods are the resources that are being managed while the work on the resource is the same. In other words, the copy_expert arguments are duplicated. Can you implement something that maybe reduces the code duplication and works on any resources?

Reply:
You're right. Quite a bit of code duplicate here. I think I'd rather just remove `_stream_and_copy()` all together because it's no longer being used.

Reply:
I don't think deleting the function is the best idea. We already put in the work to create it. Can we anticipate we'll never, ever need this functionality in the feature? I think it'd be best to put in the good work now vs. later.

Reply:
That said, please feel free to override my request. Just want to make sure my suggestion is known.

Reply:
Generally, we don't leave code in the codebase that's not being used. Because it is more code to maintain. If someone needs to reference that code or add it back in the future, the history is available in Github.

Reply:
Ok